File: backend/animal-registry/src/main/java/com/animalshelter/animalregistry/dto/MedicalRecordResponse.java
package com.animalshelter.animalregistry.dto;

import com.animalshelter.animalregistry.model.MedicalRecord;
import com.animalshelter.animalregistry.model.MedicalRecordType;

import java.time.LocalDate;
import java.time.LocalDateTime;

public record MedicalRecordResponse(
        String id,
        String animalId,
        MedicalRecordType type,
        String title,
        String description,
        LocalDate date,
        String veterinarianId,
        String veterinarianName,
        String notes,
        LocalDateTime createdAt
) {
    public static MedicalRecordResponse fromEntity(MedicalRecord record) {
        return new MedicalRecordResponse(
                record.getId(),
                record.getAnimalId(),
                record.getType(),
                record.getTitle(),
                record.getDescription(),
                record.getDate(),
                record.getVeterinarianId(),
                record.getVeterinarianName(),
                record.getNotes(),
                record.getCreatedAt()
        );
    }
}

File: frontend/src/app/dto/animal/MedicalRecordResponse.ts
import { MedicalRecordType } from './MedicalRecordType';

export interface MedicalRecordResponse {
  id: string;
  animal_id: string;
  type: MedicalRecordType;
  title: string;
  description: string | null;
  date: string;
  veterinarian_id: string;
  veterinarian_name: string;
  notes: string | null;
  created_at: string;
}

export interface MedicalRecordListResponse {
  success: boolean;
  records: MedicalRecordResponse[];
  total: number;
}
